docs/synapse/model_format.rst

```diff
@@ -0,0 +1,234 @@
+Model Definitions
+#################
+
+Quick Start
+-----------
+
+This is intended as a reference for developers working on Synapse who are creating, or adding models to the base models
+or for developers who are looking to extend Synapse for their own use by the creation of new models.
+
+This is not intended as a 'how-to' guide for Synapse hypergraph modeling, rather a document for how models may be
+defined and loaded into a cortex.
+
+Models As Code
+--------------
+
+The the core models used in Synapse hypergraph implementations are a part of the codebase which makes up Synapse.
+This allows the model definitions to grow along with the codebase over time.  One feature we have in Synapse is the
+ability to track model revisions over time. By using the Cortex extensions class, CoreModule, we can implement our
+modules, and specific handlers for them, and migration steps over time in a single place.  By placing some
+conventions around how the CoreModule is used for modeling, we can ensure consistency across both the Cortex, DataModel
+and TypeLib classes; as well as ensuring that consistent data migrations are available to Cortex users.
+
+An example of a simple model definition file is the following ::
+
+    import synapse.lib.module as s_module
+
+    def getDataModel():
```

Comment:
This convention (getDataModel) is removed from the code; just need to remove it from the docs.